spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/Adam.scala

```diff
@@ -64,22 +75,49 @@ class Adam[@specialized(Float, Double) T: ClassTag](
     val (_s, _r, _denom) =
       if (state.get[Tensor[T]]("s").isDefined) {
         (state.get[Tensor[T]]("s").get, state.get[Tensor[T]]("r").get,
-          Tensor[T]().resizeAs(dfdx).zero())
+          buffer3.resizeAs(dfdx))
       } else {
-        (Tensor[T]().resizeAs(dfdx).zero(), Tensor[T]().resizeAs(dfdx).zero(),
-          Tensor[T]().resizeAs(dfdx).zero())
+        (buffer1.resizeAs(dfdx).zero(), buffer2.resizeAs(dfdx).zero(),
```

Comment:
buffer1 and buffer2 is already saved by state table.

Reply:
ok, changed